model/histogram/histogram.go

```diff
@@ -356,26 +417,55 @@ func resize[T any](items []T, n int) []T {
 	return items[:n]
 }
 
+func clearIfNotNil[T any](items []T) []T {
```

Comment:
How about move this function into `generic.go`, since it is called both in `float_histogram.go` and `histogram.go`?